src/components/icon/js/iconService.js

```diff
@@ -422,8 +422,25 @@
     */
    function cacheIcon( id ) {
 
-     return function updateCache( icon ) {
-       iconCache[id] = isIcon(icon) ? icon : new Icon(icon, config[id]);
+     return function updateCache( _icon ) {
+       var icon = isIcon(_icon) ? _icon : new Icon(_icon, config[id]);
+
+       //clear id attributes to prevent aria issues
+       var elem = icon.element;
+       elem.removeAttribute('id');
+
+       var children = elem.childNodes;
+
+       for (var i=0; i<children.length;i++) {
+         var child = children[i];
+         if(child.nodeName==='path') {
```

Comment:
@DevVersion ie is always a pain in the neck :-)
I tried to use jsperf http://jsperf.com/remove-id-by-queryselectorall-vs-recursion but suddenly the site does not work and i cannot edit the test case. Maybe this is due to the fact that i am behind a firewall. The recursion function throwed an exception. I am pretty sure that the queryselectorall is better but i cannot prove it. May you please check on ie what is faster?